### PR TITLE
prov/efa: multi_recv test case and bug fix

### PR DESCRIPTION
**Pull request by wzamazon**
The PR contains 2 commits that are related to multi-recv:

1st commit added test case that test multi_recv of 8k message, which exposed a bug in EFA's implementation of multi reccv.

2nd commit fixed the bug.

Signed-off-by: Wei Zhang <wzam@amazon.com>

### CODE REVIEW

**Comment by wzamazon** [comment]
aws-ci is expected to fail, because the new test case exposed a bug.

File: fabtests/pytest/efa/test_multi_recv.py [hunk 1]
```diff
@@ -3,7 +3,11 @@
...
                           pytest.param("standard", marks=pytest.mark.standard)])
-def test_multi_recv(cmdline_args, iteration_type):
+@pytest.mark.parametrize("message_size", ["1024", "8192"])
+def test_multi_recv(cmdline_args, iteration_type, message_size):
     from common import ClientServerTest
-    test = ClientServerTest(cmdline_args, "fi_multi_recv -e rdm", iteration_type)
+    test = ClientServerTest(cmdline_args,
+            "fi_multi_recv -e rdm",
+            iteration_type,
+            message_size=message_size)
```
[the remark this answers is not in the document]
Comment (wzamazon) [reply fabtests/pytest/efa/test_multi_recv.py:12]: `fi_multi_recv` does not take multiple message size yet. Prior to this change, it is only run for 1k.
This change increased the coverage to both 1k and 8k

File: prov/efa/src/rxr/rxr_cq.c [hunk 2]
```diff
@@ -450,30 +451,61 @@ void rxr_cq_write_rx_completion(struct rxr_ep *ep,
...
+ *       the receiver/responder to send an RECEIPT back to sender/requester.
+ *
+ * 3. Ensure release of rx_entry.
+ *
+ * @param[in,out]	ep		endpoint
+ * @param[in,out]	rx_entry	rx entry that contains information of the RX operation
+ * @param[in]		post_ctrl	whether to post a ctrl packet back to sender/requester
+ * @param[in]		ctrl_type	ctrl packet type.
+ */
+void rxr_cq_complete_rx(struct rxr_ep *ep,
```
Comment (shijin-aws) [inline prov/efa/src/rxr/rxr_cq.c:483]: Are we going to do the same refactor for `rxr_cq_handle_tx_completion` in the future?

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_cq.c:483]: Yes.

**Pull request merged**